docs/INSTALL.md

```diff
@@ -251,6 +329,8 @@ Notes:
 
 ## AmigaOS
 
+[curl source code](https://curl.se/download.html#AmigaOS)
```

Comment:
All platforms do build from the same source tarball. It seems overkill to link to it for each individual platform. These links also point to distro endpoints, most of the time binaires, not the to the curl sources.

(this applies to all sections, with `curl source code` links.)